File: randoom/B_YetnotherrokenKeoard.java
package randoom;
import java.util.Scanner;
import java.util.Stack;

public class B_YetnotherrokenKeoard{

    static boolean isUpp(char c) {
        return c >= 'A' && c <= 'Z';
    }

    static boolean isLow(char c) {
        return c >= 'a' && c <= 'z';
    }

    static void solve() {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        int n = s.length();
        Stack<Pair<Integer, Character>> u = new Stack<>();
        Stack<Pair<Integer, Character>> l = new Stack<>();

        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            if (c == 'B') {
                if (!u.isEmpty())
                    u.pop();
            }
            if (c == 'b') {
                if (!l.isEmpty())
                    l.pop();
            }
            if (isUpp(c) && c != 'B') {
                u.push(new Pair<>(i, c));
            }
            if (isLow(c) && c != 'b') {
                l.push(new Pair<>(i, c));
            }
        }

        StringBuilder ans = new StringBuilder();
        while (!u.isEmpty() && !l.isEmpty()) {
            if (u.peek().getFirst() > l.peek().getFirst()) {
                ans.append(u.pop().getSecond());
            } else {
                ans.append(l.pop().getSecond());
            }
        }
        while (!u.isEmpty()) {
            ans.append(u.pop().getSecond());
        }
        while (!l.isEmpty()) {
            ans.append(l.pop().getSecond());
        }
        System.out.println(ans.reverse().toString());
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while (t-- > 0) {
            solve();
        }
    }

    static class Pair<F, S> {
        private F first;
        private S second;

        public Pair(F first, S second) {
            this.first = first;
            this.second = second;
        }

        public F getFirst() {
            return first;
        }

        public S getSecond() {
            return second;
        }
    }
}

File: randoom/B_YetnotherrokenKeoard.cpp
#include <iostream>
#include <stack>
#include <string>
#include <algorithm>

using namespace std;

bool isUpp(char c) {
    return c >= 'A' && c <= 'Z';
}

bool isLow(char c) {
    return c >= 'a' && c <= 'z';
}

void solve() {
    string s;
    cin >> s;
    int n = s.size();
    stack<pair<int, char>> u;
    stack<pair<int, char>> l;

    for (int i = 0; i < n; i++) {
        char c = s[i];
        if (c == 'B' && !u.empty()) {
            u.pop();
        }
        if (c == 'b' && !l.empty()) {
            l.pop();
        }
        if (isUpp(c) && c != 'B') {
            u.push({i, c});
        }
        if (isLow(c) && c != 'b') {
            l.push({i, c});
        }
    }

    string ans = "";
    while (!u.empty() && !l.empty()) {
        if (u.top().first > l.top().first) {
            ans += u.top().second;
            u.pop();
        } else {
            ans += l.top().second;
            l.pop();
        }
    }
    while (!u.empty()) {
        ans += u.top().second;
        u.pop();
    }
    while (!l.empty()) {
        ans += l.top().second;
        l.pop();
    }
    reverse(ans.begin(), ans.end());
    cout << ans << endl;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t = 1;
    cin >> t;
    while (t--) {
        solve();
    }

    return 0;
}
